### DISCUSSION TITLE
Follow up on "Example fails with minikube" issue

### DISCUSSION THREAD

**Discussion by EricBoix** [argoproj-labs/hera#1208] [Q&A] [answered]
This is a follow up of issue #58 ("Example fails with minikube")

If we look at the code refered by @flaviuvadan that is [k8s_sa.py on line 39](https://github.com/argoproj-labs/hera/blob/39f685a382cfa9f2ce2a8c77657960748502fbe6/examples/k8s_sa.py#L39C9-L39C43) (which is pretty dated code but I wasn't able to find any recent equivalent), it assumes that at least a secret will be encoutered with the expression
```python
v1.read_namespaced_service_account(self.service_account, self.namespace).secrets[0]...
```

But (by default) if you 
- start a minikube server (`minikube start`)
- install and start an argo server (e.g. with `kubectl apply -f https://raw.githubusercontent.com/argoproj/argo-workflows/master/manifests/quick-start-postgres.yaml`)
- possibly define some `argo namespace` (e.g. with `kubectl create ns argo`)
- expose the argo-server (e.g. with `kubectl -n argo port-forward service/argo-server 2746:2746 &`)
- eventually explore the available service accounts through the minikube UI (with `minikube dashboard`)

then neither of the available service accounts (`argo-server` and `argo`) have secrets defined. Which makes the `k8s_sa.py` fail when trying to retrieve the secret_name.

Is then have two questions
1. Is there some Hera on minikube installation guide (even sketchy) or examples documenting how to install argo + Hera on minikube and then get the tutorial examples to run ?
2. If the ArgoServer access is not restricted (just like the one obtained after a clean install on `minikube`), do we still need to define the [`GlobalConfig.token`](https://github.com/argoproj-labs/hera/blob/main/src/hera/shared/_global_config.py#L43) value ? (refer [here for a self-made documentation on this](https://github.com/VCityTeam/hera_utils/issues/1))

**Comment by elliotgunton** [comment, marked as answer]
Hi @EricBoix - Hera runs tests on a `k3d` K8s cluster in CI (minikube also worked when I was testing both out in https://github.com/argoproj-labs/hera/pull/887). We have a [contributors guide](https://hera.readthedocs.io/en/stable/CONTRIBUTING/#adding-new-workflow-on-cluster-tests) that explains how to use the `make` targets, which are in https://github.com/argoproj-labs/hera/blob/a787965d2901d6533f872cc8850b10a72ffda36f/Makefile#L135-L173

That might help you get set up? 

The `token` is then not needed, see the example submission [here](https://github.com/argoproj-labs/hera/blob/main/tests/submissions/test_hello_world.py).
